tempodb/backend/s3/s3_test.go

```diff
@@ -185,3 +188,39 @@ func TestObjectBlockTags(t *testing.T) {
 		})
 	}
 }
+
+func TestObjectStorageClass(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		StorageClass string
+		// expectedObject raw.Object
+	}{
+		{
+			"Standard", "STANDARD",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// rawObject := raw.Object{}
+			var obj url.Values
+
+			server := fakeServerWithHeader(STORAGECLASSHEADER, t, &obj)
+			_, w, _, err := New(&Config{
+				Region:       "blerg",
+				AccessKey:    "test",
+				SecretKey:    flagext.SecretWithValue("test"),
+				Bucket:       "blerg",
+				Insecure:     true,
+				Endpoint:     server.URL[7:], // [7:] -> strip http://
+				StorageClass: tc.StorageClass,
+			})
+			require.NoError(t, err)
+
+			ctx := context.Background()
+			_ = w.Write(ctx, "object", backend.KeyPath{"test"}, bytes.NewReader([]byte{}), 0, false)
```

Comment:
Does this test check the storage class was actually used?  Comparing to `TestObjectBlockTags` which looks at the results in `obj`.

Reply:
Good point